frontend/src/components/Link/KialiPageLink.tsx

```diff
@@ -20,11 +20,7 @@ class KialiPageLinkComponent extends React.Component<KialiPageLinkProps> {
   render() {
     // Without a cluster, simply render a local link
     // If cluster is specified, and it's the home cluster, render a local link.
-    if (
-      !this.props.cluster ||
-      !serverConfig.clusterInfo?.name ||
-      this.props.cluster === serverConfig.clusterInfo.name
-    ) {
+    if (!this.props.cluster || homeCluster?.name || this.props.cluster === homeCluster?.name) {
```

Comment:
```suggestion
    if (!this.props.cluster || !homeCluster?.name || this.props.cluster === homeCluster.name) {
```

I think you missed a negation?

Reply:
This is actually somewhat confusing to me. This is testing `homeCluster?.name !== ""`?

Reply:
I guess if `homeCluster` is undefined then `!undefined` would also evaluate to `true`.

Reply:
I'll be honest, I didn't overthink my comment, I was just looking at the original code and it looked like we dropped the negation on the middle expression.

Reply:
I actually dropped two conditionals. Added them back in latest commit and now the FE unit tests are now passing.